.github/workflows/build-artifact.yaml

```diff
@@ -57,15 +58,27 @@ jobs:
       - name: Update submodules
         run: |
           git submodule update --init --recursive
+      - name: Cache build artifacts
+        id: cache-build-artifacts
+        uses: actions/cache@v4
+        with:
+          path: build/
+          key: build-${{ inputs.build-type }}-${{ matrix.arch }}-${{ inputs.tracy && 'tracy' || 'notracy' }}-${{ github.sha }}
```

Comment:
how does `${{ github.sha }}` work in this case? I assume we'll need to pull the cache from a different github sha than what the workflow is currently triggered on?

Reply:
Plan is to use a key without `github.sha` as a restore key - meaning that if it can't find the exact key match, it will default to `build-${{ inputs.build-type }}-${{ matrix.arch }}-${{ inputs.tracy && 'tracy' || 'notracy' }}-`. I haven't updated the PR for this yet.

Since current plan is to only save caches from main branch, it should pull most recent cache from main branch with prefix `build-${{ inputs.build-type }}-${{ matrix.arch }}-${{ inputs.tracy && 'tracy' || 'notracy' }}-`.

Reply:
nice!

Reply:
another option to consider is to use replace-keys for 'partial cache hits' according to keys
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows

Reply:
Mmmm do you mean `restore-keys`? @ttmch